### PR TITLE
Yandex.Disk module implemented to support Yandex.Disk cloud storage

### PR DESCRIPTION
**Pull request by vpotapev**
This is a module for supporting [Yandex.Disk](https://disk.yandex.com/) - a cloud storage created by [Yandex](https://en.wikipedia.org/wiki/Yandex).

### CODE REVIEW

File: py3status/modules/yandexdisk_status.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    format = 'Yandex.Disk: {}'
+
+    def yadisk(self, i3s_output_list, i3s_config):
+        response = {'cached_until': time() + self.cache_timeout}
+
+        raw_lines = ''
+        try:
+            raw_lines = subprocess.check_output(shlex.split('yandex-disk status'))
+        except subprocess.CalledProcessError as e:
+            if e.returncode == 1:
```
Comment (pydsigner) [inline py3status/modules/yandexdisk_status.py]: If `e.returncode != 1`, we'll end up getting a NameError at line 43.

Comment (vpotapev) [inline py3status/modules/yandexdisk_status.py]: @pydsigner 
If `e.returncode != 1`, the raw_lines variable will be empty string (it is defined at line 36). So it seems it should work correctly. But maybe I've missed something, and you have meant another problem?

Comment (pydsigner) [inline py3status/modules/yandexdisk_status.py]: Ah you're right, I missed that.

File: py3status/modules/yandexdisk_status.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+Display Yandex.Disk status.
+
+Configuration parameters:
+    cache_timeout: how often we refresh this module in seconds (10s default)
+    format: prefix text for the Yandex.Disk status
+
+Valid status values include:
+    - Yandex.Disk isn't running!
+    - Error: daemon not started
+    - ...
```
Comment (tobes) [inline py3status/modules/yandexdisk_status.py]: This seems irrelevant as the user of this module will not see these.  Maybe just saying you can tell it is running etc in the main docstring.

File: py3status/modules/yandexdisk_status.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import subprocess
+from time import time
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    cache_timeout = 10
+    format = 'Yandex.Disk: {}'
```
Comment (tobes) [inline py3status/modules/yandexdisk_status.py]: `{}` should be `{status}` or similar.

File: py3status/modules/yandexdisk_status.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    cache_timeout: how often we refresh this module in seconds (10s default)
+    format: prefix text for the Yandex.Disk status
+
+Valid status values include:
+    - Yandex.Disk isn't running!
+    - Error: daemon not started
+    - ...
+
+Requires:
+    Yandex.Disk CLI: command line tool (link: https://disk.yandex.com/)
```
Comment (tobes) [inline py3status/modules/yandexdisk_status.py]: I'd use the actual needed program `yandex-disk`

File: py3status/modules/yandexdisk_status.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        response = {'cached_until': time() + self.cache_timeout}
+
+        raw_lines = ''
+        try:
+            raw_lines = subprocess.check_output(shlex.split('yandex-disk status'))
+        except subprocess.CalledProcessError as e:
+            if e.returncode == 1:
+                raw_lines = e.output
+
+        lines = raw_lines.decode('utf-8').split('\n')
```
Comment (tobes) [inline py3status/modules/yandexdisk_status.py]: if raw lines is `''` line 36 will this not die in python3?

Comment (pydsigner) [inline py3status/modules/yandexdisk_status.py]:
```pytb
>>> ''.decode('utf-8')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'str' object has no attribute 'decode'
```

File: py3status/modules/yandexdisk_status.py (hunk 2 above)
Comment (vpotapev) [inline py3status/modules/yandexdisk_status.py]: @tobes Agree. I've moved it into main docstring of the module.

File: py3status/modules/yandexdisk_status.py (hunk 5 above)
Comment (vpotapev) [inline py3status/modules/yandexdisk_status.py]: @tobes @pydsigner Fixed.

File: py3status/modules/yandexdisk_status.py (hunk 3 above)
Comment (vpotapev) [inline py3status/modules/yandexdisk_status.py]: @tobes Done.

File: py3status/modules/yandexdisk_status.py (hunk 4 above)
Comment (vpotapev) [inline py3status/modules/yandexdisk_status.py]: @tobes Done.

File: py3status/modules/yandexdisk_status.py (hunk 2 above)
Comment (tobes) [inline py3status/modules/yandexdisk_status.py]: Sorry, maybe I wasn't clear.

The docstring for the module (the one we are in here) should have all the info.  stuff like 'Configuration parameters:'  __must__ be in it. This should be written in a style that users can understand, including people who have no idea what the module is.

README.md is generated by `py3status docstring update <module>`

File: py3status/modules/yandexdisk_status.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+            status = 'Not started'
+            response['color'] = i3s_config['color_bad']
+        elif status == "Synchronization core status: idle":
+            status = 'Idle'
+            response['color'] = i3s_config['color_good']
+        else:
+            status = 'Busy'
+            response['color'] = i3s_config['color_degraded']
+
+        full_text = self.format.format(status=str(status))
```
Comment (tobes) [inline py3status/modules/yandexdisk_status.py]: is there a reason for the `str(...)`?

Comment (pydsigner) [inline py3status/modules/yandexdisk_status.py]: @tobes `.format()` should definitely take care of that.

File: py3status/modules/yandexdisk_status.py (hunk 2 above)
Comment (vpotapev) [inline py3status/modules/yandexdisk_status.py]: @tobes 
I'ts a bit strange but when I run `py3status docstring update yandexdisk_status` it says me `py3status: Setup error (IOError) docstrings.py line 26`.

At this line: `readme_file = os.path.join(modules_directory(), 'README.md')`
I've printed the `readme_file` value: `/usr/local/lib/python2.7/dist-packages/py3status-2.9-py2.7.egg/py3status/modules/README.md`. But there is no file with this name. I'm not sure but maybe it's a bug.

Comment (tobes) [inline py3status/modules/yandexdisk_status.py]: @vpotapev you need to be running the full thing ie clone of the github repo.  It only really makes sense for development.

Comment (vpotapev) [inline py3status/modules/yandexdisk_status.py]: @tobes
But I have it. Repo was cloned (`git clone URL`). I've made a build (`python ./setup.py clean build`), installed it (`sudo python ./setup.py install`). After that I've restarted my i3. At this point I have the py3status fresh installed built from repo. And then I get this error above.
Actually I thought that the `docstring update` command will work as the Doxygen works and documentation updates in current development directory, so it's strange to see that `docstring update` tries to find somethin in already installed software at /usr/local/dir.
I'll try to find out how it can be done, it's just looks unusual.

Comment (tobes) [inline py3status/modules/yandexdisk_status.py]: > sudo python ./setup.py install

use `python ./setup.py develop`

http://pythonhosted.org//setuptools/setuptools.html#development-mode

**Comment by ultrabug** [comment]
Thanks a lot for your contribution @vpotapev !

Reviewers, are we close enough for a merge in your opinion ?

**Comment by tobes** [comment]
> are we close enough for a merge 

The `format` issue on line 58 and the doctstring could use some work. but nothing that I'd say is blocking.

**Comment by ultrabug** [comment]
Thanks a lot @vpotapev !

@tobes if you're up for it, plz implement the formatting cleanup, cheers !
